opr/grading.go

```diff
@@ -70,52 +71,17 @@ func GradeBlock(list []*OraclePriceRecord) (tobepaid []*OraclePriceRecord, sorte
 	}
 	list = nlist
 
-	last := len(list)
 	// Throw away all the entries but the top 50 in difficulty
-	// bubble sort because I am lazy.  Could be replaced with about anything
-	for j := 0; j < len(list)-1; j++ {
-		for k := 0; k < len(list)-j-1; k++ {
-			d1 := list[k].Difficulty
-			d2 := list[k+1].Difficulty
-			if d1 == 0 || d2 == 0 {
-				//panic("Should not be here")
-			}
-			if d1 < d2 { // sort the smallest difficulty to the end of the list
-				list[k], list[k+1] = list[k+1], list[k]
-			}
-		}
-	}
+	// Note that we are sorting in descending order.
+	sort.Slice(list,func(i,j int)bool{return list[i].Difficulty>list[j].Difficulty})
```

Comment:
should this be `sort.SliceStable` to guarantee stability? it seems to work out ok in all the situations but there might be some edge cases where it doesn't, or the underlying implementation might change.

Reply:
That sort has no need to preserve any prior ordering.  The observation is that difficulties will never match (odds are about 1 in 2^(64-#leading matching bits)).  This would be mitigated by doing the second sort as a standard quicksort, but also mitigated by a stable sort.  The slight performance improvement is likely worth the risk?  But this routine isn't in our inner loop so its performance isn't really interesting.

Reply:
I don't think there is any measurable gain to changing the sort, but also no real gain in not changing the sort.  I think it is your call.

Reply:
I agree, with both of you. But if I had to really decided, I would have a preference with the stable sort.

Reply:
i didn't consider the real life values of the Difficulty field, i was just looking at test data with a lot of potential collisions. i'll probably do some benchmarks to see if there's a noticeable performance difference but otherwise i'd say make it a stablesort